tools/src/linux_commands.rs

```diff
@@ -46,26 +46,30 @@ pub fn write_new_file(path: impl AsRef<Path>, content: impl AsRef<[u8]>) -> Resu
     }
 }
 
-/// Generates a private key, and returns it in PEM format.
-/// Writes PEM to `output_file`.
-/// Returns error if `output_file` already exists.
-pub fn create_private_key_pem(output_file: impl AsRef<Path>) -> Result<String> {
-    let privkey_pem = execute_and_parse_stdout(
-        Command::new("openssl")
-            .arg("ecparam")
-            .arg("-outform")
-            .arg("pem")
-            .arg("-name")
-            .arg("prime256v1")
-            .arg("-genkey"),
-    )?;
-    write_new_file(output_file, &privkey_pem)?;
-    Ok(privkey_pem)
+/// Tries to read the content of given file; if the file does not exist,
+/// generates a private key, and writes PEM to file, and returns it.
+pub fn read_or_create_private_key_pem(file: impl AsRef<Path>) -> Result<String> {
+    if file.as_ref().exists() {
+        std::fs::read_to_string(file).map_err(Error::new)
+    } else {
+        let privkey_pem = execute_and_parse_stdout(
+            Command::new("openssl")
+                .arg("ecparam")
+                .arg("-outform")
+                .arg("pem")
+                .arg("-name")
+                .arg("prime256v1")
+                .arg("-genkey"),
+        )
+        .map_err(|e| e.context("Failed to use openssl to generate private key"))?;
```

Comment:
As an aside, please write up how the user can extract the private key for safe keeping.  After the registration, the private key will be important in future cert requests.